src/main/java/view/panels/PortfolioPanel.java

```diff
@@ -58,34 +61,48 @@ private JTable createPortfolioTable() {
         return table;
     }
 
-    private DefaultTableModel getDefaultTableModel() {
-        String[] columnNames = {
-                "Ticker", "Company Name", "Average Cost", "Quantity",
-                "Market Price", "Profit / Share", "Total Profit"
-        };
-
-        Object[][] data = {
-                {"NVDA", "NVIDIA Corporation", "138.27", "2", "158.03", "19.76", "40.00"},
-                {"TSLA", "Tesla, Inc.", "110.23", "12", "101.85", "-8.38", "-100.56"},
-                {"INTC", "Intel Corporation", "XX.XX", "X", "XX.XX", "XX.XX", "XX.XX"},
-        };
-
-        return new DefaultTableModel(data, columnNames) {
-            @Override
-            public boolean isCellEditable(int row, int column) {
-                return false;
-            }
-        };
+    private String[] getColumnNames() {
+        return new String[]{"Ticker", "Average Cost", "Quantity", "Market Price", "Profit / Share", "Total Profit"};
     }
 
     @Override
     public void receiveViewEvent(ViewEvent event) {
-        // Future handling of UpdatePortfolioEvent
+        if (event instanceof UpdateAssetEvent updateEvent) {
```

Comment:
Since we are already using Java 17, we can try to use pattern matching with `switch` statement instead of if-else for better readability. For example:
```python
public void receiveViewEvent(ViewEvent event) {
    switch (event) {
        case UpdateAssetEvent updateEvent -> {
            // Get the portfolio from the event
            Portfolio portfolio = updateEvent.getPortfolio();

            // Clear the current table content
            tableModel.setRowCount(0);

            // Populate table with updated data from the portfolio
            portfolio.getAllStocks().forEach(userStock -> {
                Object[] rowData = createRowData(userStock);
                tableModel.addRow(rowData);
            });
        }
    }
}
```

Reply:
I just tried it and java pattern matching requires a `default` clause for pattern matching. In `default`, simply do nothing as the event is not used for this panel.

Reply:
I'm considering again that since there are only one or two events, it is not necessary to use pattern matching here. Keep the code as what it is now.